walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,169 @@
+      //    |----ending {@code TimeRange}----| available time |----starting {@code TimeRange}----|
+      int availableTimeRangeStart = endingTimeRange.end();
+      int availableTimeRangeEnd = startingTimeRange.start();
+      checkDurationAndAddAvailableTimeRange(requestedDuration,
+                                            availableTimeRangeStart, availableTimeRangeEnd,
+                                            availableTimeRanges);
+
+      // Find the potential, next block of available time sandwiched between an ending and a
+      // starting {@code TimeRange}.
+      // (We can safely increment both pointers without missing any available time, because
+      //  we had the guarantee that no other existing {@code TimeRange} can be found between
+      //  the current ending and starting {@code TimeRange}.)
+      if (startOrderedTimeRangesIterator.hasNext() && endOrderedTimeRangesIterator.hasNext()) {
+        startingTimeRange = startOrderedTimeRangesIterator.next();
+        endingTimeRange = endOrderedTimeRangesIterator.next();
+      } else {
+        break;
+      }
+    }
+
+    // Potentially add the {@code TimeRange} after the last existing {@code TimeRange}.
+    TimeRange lastEndingTimeRange = endOrderedTimeRanges.get(endOrderedTimeRanges.size() - 1);
+    checkDurationAndAddAvailableTimeRange(requestedDuration,
+                                          lastEndingTimeRange.end(), TimeRange.END_OF_DAY + 1,
+                                          availableTimeRanges);
+
+    return availableTimeRanges;
+  }
+
+  /**
+  * Finds the existing {@code TimeRange} that correspond to {@code Event} having
+  * overlapping attendees with a set of attendees that we are concerned with.
+  */
+  private Collection<TimeRange> getRelevantTimeRangesFromEvents(Collection<Event> events,
+      Set<String> attendees) {
+    Collection<TimeRange> relevantTimeRanges = new HashSet<>();
+    for (Event event : events) {
+      Set<String> overlappingAttendees = new HashSet<>(event.getAttendees());
+      overlappingAttendees.retainAll(attendees);
+      if (!overlappingAttendees.isEmpty()) {
+        relevantTimeRanges.add(event.getWhen());
+      }
+    }
+    return relevantTimeRanges;
+  }
+
+  /**
+  * Sorts a collection of {@code TimeRange} based on the specified {@code sortOrder}.
+  */
+  private List<TimeRange> getOrderedTimeRanges(Collection<TimeRange> timeRanges,
+      Comparator<TimeRange> sortOrder) {
+    List<TimeRange> orderedTimeRanges = new LinkedList<>(timeRanges);
+    Collections.sort(orderedTimeRanges, sortOrder);
+    return orderedTimeRanges;
+  }
+
+  /**
+  * If a [start, end) represented {@code TimeRange} has a sufficient duration, adds it to the
+  * set of available {@code TimeRange}.
+  */
+  private void checkDurationAndAddAvailableTimeRange(long requestedDuration, int start, int end,
+      Collection<TimeRange> availableTimeRanges) {
+    if (Long.compare(requestedDuration, end - start) <= 0) {
```

Comment:
Since requestDuration, end, and start are all primitive types, I don't think you need to use the Long class to compare them (i.e. can just do requestedDuration <= end - start)